Origin CA certificate validity not applied (provider version >= `v2.22.0`)

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v0.14.11
Cloudflare Provider `>= v2.22.0`

### Affected resource(s)

cloudflare_origin_ca_certificate

### Terraform configuration files

```hcl
resource "tls_private_key" "example" {
  algorithm = "RSA"
}

resource "tls_cert_request" "example" {
  key_algorithm   = tls_private_key.example.algorithm
  private_key_pem = tls_private_key.example.private_key_pem

  subject {
    common_name  = ""
    organization = "Terraform Test"
  }
}

resource "cloudflare_origin_ca_certificate" "example" {
  csr                = tls_cert_request.example.cert_request_pem
  hostnames          = [ "example.com" ]
  request_type       = "origin-rsa"
  requested_validity = 30
}
```


### Debug output

Should be reproducible by running Terraform acceptance tests?

### Panic output

_No response_

### Expected output

Expect a certificate to be created that is valid for 30 days

### Actual output

Certificate that is created is valid for 5475 days

### Steps to reproduce

The easiest way would be to run the Terraform acceptance tests within this repository. I was going to extend the tests to showcase this but it looks like they are already failing:

```
=== RUN   TestAccCloudflareOriginCACertificate_Basic
    testing.go:705: Step 0 error: Check failed: Check 6/6 error: cloudflare_origin_ca_certificate.hfjigatemd: Attribute 'requested_validity' expected "7", got "5475"
--- FAIL: TestAccCloudflareOriginCACertificate_Basic (7.76s)
FAIL
exit status 1
FAIL    github.com/cloudflare/terraform-provider-cloudflare/cloudflare  7.767s
``` 

### Additional factoids

_No response_

### References

Issue was introduced with change in #1078

--- a/.changelog/1214.txt
+++ b/.changelog/1214.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/cloudflare_origin_ca_certificate: ignore `requested_validity` changes due to the value decreasing but still store it
+```

--- a/cloudflare/resource_cloudflare_origin_ca_certificate.go
+++ b/cloudflare/resource_cloudflare_origin_ca_certificate.go
@@ -56,10 +56,8 @@ func resourceCloudflareOriginCACertificate() *schema.Resource {
 			"requested_validity": {
 				Type:         schema.TypeInt,
 				Optional:     true,
+				Computed:     true,
 				ValidateFunc: validation.IntInSlice([]int{7, 30, 90, 365, 730, 1095, 5475}),
-				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
-					return true
-				},
 			},
 		},
 	}
@@ -83,12 +81,11 @@ func resourceCloudflareOriginCACertificateCreate(d *schema.ResourceData, meta in
 		certInput.CSR = csr.(string)
 	}
 
-	requestValidity, ok := d.GetOk("requested_validity")
-	if ok {
+	if requestValidity, ok := d.GetOk("requested_validity"); ok {
 		certInput.RequestValidity = requestValidity.(int)
 	}
 
-	log.Printf("[INFO] Creating Cloudflare OriginCACertificate: hostnames %v", hostnames)
+	log.Printf("[INFO] Creating Cloudflare OriginCACertificate: %#v", certInput)
 	cert, err := client.CreateOriginCertificate(context.Background(), certInput)
 
 	if err != nil {

